test/jdk/java/util/regex/NamedGroupsTests.java

```diff
@@ -0,0 +1,277 @@
+
+    public static void main(String[] args) {
+        testPatternNamedGroups();
+        testMatcherNamedGroups();
+        testMatchResultNamedGroups();
+
+        testMatcherHasMatch();
+        testMatchResultHasMatch();
+
+        testMatchResultStartEndGroup();
+    }
+
+    private static void testMatchResultStartEndGroup() {
+        testMatchResultStartEndGroup1();
+        testMatchResultStartEndGroup2();
+        testMatchResultStartEndGroup3();
+        testMatchResultStartEndGroup4();
+    }
+
+    private static void testMatchResultStartEndGroup1() {
+        List<MatchResult> list = Pattern.compile("(?<some>.+?)(?<rest>.*)").matcher("")
+                .results().toList();
+        for (var result : list) {
+            if (result.start("some") >= 0) {
+                throw new RuntimeException("start(\"some\")");
+            }
+            if (result.start("rest") >= 0) {
+                throw new RuntimeException("start(\"rest\")");
+            }
+            if (result.end("some") >= 0) {
+                throw new RuntimeException("end(\"some\")");
+            }
+            if (result.end("rest") >= 0) {
+                throw new RuntimeException("end(\"rest\")");
+            }
+            if (result.group("some") != null) {
+                throw new RuntimeException("group(\"some\")");
+            }
+            if (result.group("rest") != null) {
+                throw new RuntimeException("group(\"rest\")");
+            }
+        }
+    }
+
+    private static void testMatchResultStartEndGroup2() {
+        List<MatchResult> list = Pattern.compile("(?<some>.+?)(?<rest>.*)").matcher("abc")
+                .results().toList();
+        for (var result : list) {
+            if (result.start("some") < 0) {
+                throw new RuntimeException("start(\"some\")");
+            }
+            if (result.start("rest") < 0) {
+                throw new RuntimeException("start(\"rest\")");
+            }
+            if (result.end("some") < 0) {
+                throw new RuntimeException("end(\"some\")");
+            }
+            if (result.end("rest") < 0) {
+                throw new RuntimeException("end(\"rest\")");
+            }
+            if (result.group("some") == null) {
+                throw new RuntimeException("group(\"some\")");
+            }
+            if (result.group("rest") == null) {
+                throw new RuntimeException("group(\"rest\")");
+            }
+        }
+    }
+
+    private static void testMatchResultStartEndGroup3() {
+        Pattern.compile("(?<some>.+?)(?<rest>.*)").matcher("")
+                .results()
+                .forEach(r -> {
+                        try {
+                            r.start("noSuchGroup");
+                            r.end("noSuchGroup");
+                            r.group("noSuchGroup");
+                        } catch (IllegalArgumentException e) {  // swallowing intended
+                        }
```

Comment:
If MatchResult is behind correctly, the call to `r.start("noSuchGroup")` will always throw an exception and the subsequent calls to `r.end` and `r.group` will never be executed. This potentially will miss testing of those methods.